### ISSUE TITLE
Use OsString instead of String and OsStr instead of str

### ISSUE THREAD

**Issue by haraldh** [kpcyrd/kmod-rs#9]
While it is very convenient to use the standard `str` and `String` it doesn't feel right to do s.th. like
```
path.to_string_lossy().into_owned()
```
especially if the the path can contain non UTF-8 conform bytes and you really want to use this path later on (not just for printing) in a `PathBuf`.

**Comment by twistedfall** [comment]
I think this one is also kind of fixed by https://github.com/kpcyrd/kmod-rs/pull/10. The thing is you can't really return `OsStr` easily because we only get `CStr` from ffi api. And to get from `CStr` to `OsStr` reliably you'd still need to go through `&str`. So now we just return `&str` and let the caller deal with it. But at least there is no lossy conversion and no allocation now.

**Comment by haraldh** [comment]
```rust
    #[inline]
    pub fn path(&self) -> Option<&OsStr> {
        use std::os::unix::ffi::OsStrExt;

        unsafe {
            kmod_sys::kmod_module_get_path(self.inner)
                .as_ref()
                .map(|ptr| CStr::from_ptr(ptr))
        }
        .map(CStr::to_bytes)
        .map(OsStr::from_bytes)
    }
```

**Comment by twistedfall** [comment]
Well, [OsStr::from_bytes](https://doc.rust-lang.org/stable/std/os/unix/ffi/trait.OsStrExt.html#tymethod.from_bytes) is a available on Unix only. Yes, one might argue that `kmod` is only available on Unix-like (or even only on Linux), yet it will still make the code not clean. E.g. without conditional `#[cfg]`s you won't be able to compile some windows/mac os binary which depends on `kmod` even if the code in question will not be used.

**Comment by haraldh** [comment]
There is no kmod-sys for windows either.

**Comment by twistedfall** [comment]
That's true. Ultimately I don't really mind changing it to `OsStr`, can't speak for the crate maintainer. I suppose doing a pull request would be the best.

**Comment by kpcyrd** [comment]
I think this been solved with #14
